rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -1221,10 +1221,8 @@ bool PlayerImpl::publish_message(rosbag2_storage::SerializedBagMessageSharedPtr
 
   // Try to publish message as service request
   auto client_iter = service_clients_.find(message->topic_name);
-  if (client_iter != service_clients_.end()) {
+  if (play_options_.publish_service_requests && client_iter != service_clients_.end()) {
```

Comment:
I think `play_options_.publish_service_requests` is unnecessary.   
In the process of PlayerImpl::prepare_publishers(), `play_options_.publish_service_requests` can make sure service event topic is in publishers_.   
So if play_options_.publish_service_requests is not set, the service event topic is already handled in the code for publisher_.

Reply:
@Barry-Xu-2018 I would keep it for defensive programming or if in the future one will decide to change the value of the `play_options_.publish_service_requests` in runtime.